homeassistant/components/bluesound/media_player.py

```diff
@@ -314,6 +315,15 @@ def __init__(
 
         self._init_callback = init_callback
 
+        self._attr_unique_id = format_unique_id(sync_status.mac, port)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, format_unique_id(sync_status.mac, port))},
```

Comment:
We ideally want to have 1 device for 1 physical device. (There are exceptions for for example climate entities that create a climate entity per room). Would it be better if we use the `connections` field and set the Mac address to the Mac and if someone has a device with the virtual port, it would merge those two devices.

Now that i think of it, maybe that's not a good approach, I'm not sure, what do you think of this and can you maybe elaborate on how the different virtual devices are used?

Reply:
This is from the documentation:

> port 11000 is used for all BluOS players, with the exception
> of the CI580. The CI580 has four streamers nodes in one chassis, where node 1 uses port 11000, node 2
> uses port 11010, node 3 uses port 11020, and node 4 uses port 11030. The actual port to use should be
> discovered by use of the MDNS protocol using the services musc.tcp and musp.tcp.

This seems to indicate that each port is like an independent player(`four streamer nodes in one chassis`). The CI580 is a rack mounted device and has 4 separate stereo zones. I would assume that each of those zones is used for a different room. So this is probably similar to `climate per room`.

I also think this is a very rare edge case. Not many people will use such a device.